### PR TITLE
Fix #148 - In case underlying JDBC doesn't return the Statement assoc…

### LINKED ISSUE #148: Anorm AkkaStream throwing NPE
### Are you looking for help?

I posted this issue in Play Framework Google Group but didn't get a response.

https://groups.google.com/forum/#!topic/play-framework/OIscPmFpH5I

### Anorm Version (2.5.x / etc)

2.5.3

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)

Linux ubuntu 4.4.0-66-generic #87-Ubuntu SMP Fri Mar 3 15:29:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)

openjdk version "1.8.0_121"
OpenJDK Runtime Environment (build 1.8.0_121-8u121-b13-0ubuntu1.16.04.2-b13)
OpenJDK 64-Bit Server VM (build 25.121-b13, mixed mode)

### Library Dependencies

Akka Streams version: 2.4.16
Play version: 2.4.3
DB: MySQL 5.6 on Amazon RDS
mysql-connector-java version: 5.1.38

### Expected Behavior

1. Materialize a source with Akka Streams Anorm integration.

### Actual Behavior

I got the following exception when running an Akka Stream using Anorm as the Source. I've seen this happen twice so far. Most of the time it works.
```
java.lang.NullPointerException: null
        at anorm.AkkaStream$ResultSource$$anon$1$$anonfun$release$1.apply(AkkaStream.scala:128)
        at anorm.AkkaStream$ResultSource$$anon$1$$anonfun$release$1.apply(AkkaStream.scala:127)
        at scala.Option.foreach(Option.scala:257)
        at anorm.AkkaStream$ResultSource$$anon$1.release(AkkaStream.scala:127)
        at anorm.AkkaStream$ResultSource$$anon$1.postStop(AkkaStream.scala:116)
        at akka.stream.impl.fusing.GraphInterpreter.finalizeStage(GraphInterpreter.scala:801)
        at akka.stream.impl.fusing.GraphInterpreter.afterStageHasRun(GraphInterpreter.scala:780)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:664)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:471)
        at akka.stream.impl.fusing.GraphInterpreterShell.init(ActorGraphInterpreter.scala:381)
        at akka.stream.impl.fusing.ActorGraphInterpreter.tryInit(ActorGraphInterpreter.scala:538)
        at akka.stream.impl.fusing.ActorGraphInterpreter.preStart(ActorGraphInterpreter.scala:586)
        at akka.actor.Actor$class.aroundPreStart(Actor.scala:504)
        at akka.stream.impl.fusing.ActorGraphInterpreter.aroundPreStart(ActorGraphInterpreter.scala:529)
        at akka.actor.ActorCell.create(ActorCell.scala:590)
        at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:461)
        at akka.actor.ActorCell.systemInvoke(ActorCell.scala:483)
        at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:282)
        at akka.dispatch.Mailbox.run(Mailbox.scala:223)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

My use case is very basic:
```scala
  def taskSource()(implicit m: Materializer): Source[TASK, Future[Int]]#ReprMat[Task, Unit] = {
    
    implicit val connection = db.getConnection()
    
    val sql = SQL("SELECT * FROM tasks")
    AkkaStream.source(sql, Task.parser).mapMaterializedValue( _.onComplete { _ =>
      connection.close()
    })
  }
```

I believe the problematic code is that ResultSet.getStatement() is returning null (not sure how this happens), and then the null is wrapped in a Some() and hence the NPE happens.
```scala
        override def postStop() = release()

        private def release() {
          val stmt: Option[java.sql.Statement] = {
            if (resultSet != null && !resultSet.isClosed) {
              val s = resultSet.getStatement
              resultSet.close()
              Some(s)                  // <---- s is null somehow
            } else None
          }

          stmt.foreach { s =>
            if (!s.isClosed) s.close() // <---- NPE thrown on this line
          }
        }
```

The Some(s) should be changed to an Option(s) since s could possibly be null.

### Reproducible Test Case

Please provide a PR with a failing test.  

If the issue is more complex or requires configuration, please provide a link to a project on Github that reproduces the issue.

### PR DESCRIPTION
**Pull request by cchantep**
…iated with the current ResultSet

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -102,7 +102,7 @@ lazy val `anorm-iteratee` = (project in file("iteratee"))
     ) ++ specs2Test
   )).dependsOn(anorm)
 
-val akkaVer = "2.4.17"
+val akkaVer = "2.4.13"
```
Comment (richdougherty) [inline build.sbt]: Is there a reason you're using an older Akka version here?

Comment (cchantep) [inline build.sbt]: typo fixed

Comment (richdougherty) [inline build.sbt]: OK :)

**Comment by cchantep** [comment]
Thx @richdougherty

**Comment by richdougherty** [comment]
No problem. :)

**Pull request merged**